CRM/Report/Form/Event/Income.php

```diff
@@ -122,7 +121,7 @@ public function buildEventReport($eventIDs) {
             LEFT JOIN  civicrm_participant ON ( civicrm_event.id = civicrm_participant.event_id
                        {$activeParticipantClause} AND civicrm_participant.is_test  = 0 )
 
-            WHERE      civicrm_event.id IN( {$eventID}) {$groupBy}";
+            WHERE      civicrm_event.id IN( {$eventID}) GROUP BY civicrm_event.id";
```

Comment:
This will lead to full group by DB error on MySql 5.7+. Can you modify `$groupBy` line to this and check if it fixes the bug?

    $groupBy = CRM_Contact_BAO_Query::getGroupByFromSelectColumns($select, 'civicrm_event.id');

Thanks!